lib/segment/src/types.rs

```diff
@@ -99,6 +99,8 @@ impl FromStr for DateTimePayloadType {
     fn from_str(s: &str) -> Result<Self, Self::Err> {
         // Attempt to parse the input string in RFC 3339 format
         if let Ok(datetime) = chrono::DateTime::parse_from_rfc3339(s)
+            // Attempt to parse default to-string format
+            .or_else(|_| chrono::DateTime::from_str(s))
```

Comment:
According to [`chrono` documentation][chrono-date-time-from-str], `FromStr` implementation:

> Accepts a relaxed form of RFC3339. A space or a ‘T’ are accepted as the separator between the date and time parts. Additional spaces are allowed between each component.

Which seems exactly what we want here? Maybe we can even put this as the first parser we try, before `parse_from_rfc3339`.

[chrono-date-time-from-str]: https://docs.rs/chrono/latest/chrono/struct.DateTime.html#impl-FromStr-for-DateTime%3CFixedOffset%3E